assets/training/aoai/proxy_components/src/common/cancel_handler.py

```diff
@@ -4,19 +4,30 @@
 """Cancel handler class."""
 
 import signal
+from proxy_component import AzureOpenAIProxyComponent
+from common.logging import get_logger, add_custom_dimenions_to_app_insights_handler
+logger = get_logger(__name__)
 
 
 class CancelHandler:
     """Cancel handler."""
 
-    cancel_triggered = False
-
-    def __init__(self):
+    def __init__(self, component: AzureOpenAIProxyComponent):
         """Create CancelHandler class."""
+        self.component = component
         signal.signal(signal.SIGINT, self.cancel)
         signal.signal(signal.SIGTERM, self.cancel)
 
     def cancel(self, *args):
         """Cancel method to handle the signal."""
-        print('cancel triggered')
-        self.cancel_triggered = True
+        if self.component is not None:
+            logger.info("calling cancel job for the component")
+            self.component.cancel_job()
+        else:
+            logger.warning("component is none for the cancel handler")
+
+    def register_cancel_handler(component: AzureOpenAIProxyComponent):
+        """Register component cancel job method."""
+        add_custom_dimenions_to_app_insights_handler(logger, component)
```

Comment:
why we do this here?

Reply:
this is to add fields in custom dimension, we can do it in constructor as well.